packages/react-popover/src/components/Popover/usePopover.ts

```diff
@@ -95,6 +109,13 @@ function usePopoverRefs(state: PopoverState): PopoverState {
     state.offset = getOffsetWithArrow(state.offset, arrowHeights[state.size]);
   }
 
+  const [contextTarget, setContextTarget] = React.useState<PopperVirtualElement>();
+  state.setContextTarget = setContextTarget;
+
+  if (!state.target && state.openOnContext) {
+    state.target = contextTarget as HTMLElement;
+  }
```

Comment:
We discussed this in offline: we should reset `contextTarget` for other event as it causes a problem with invisible Popover. To repro:
- open a default story
- enable `openOnContext`
- click on a button with a right mouse button (✅ Popover is visible)
- click outside with a mouse (✅ Popover is hidden)
- click on a button with left mouse button (❌ Popover is hidden)

Reply:
We should discuss with a wider audience if we should support both left+right click at the same time out of the box

Reply:
But I will address the issue that the target is not being set correctly now

Reply:
hover+context will work correctly now and the target will update with right clock correctly

Added an agenda item to weekly tech sync to ask about left+right click at the same time, resolving for now